fix: BROS-49: TaskSummary postfixes

niklub opened the pull request:
- Display predictions
- Allow to resize columns
- Sort columns so global classifications are going first, then labels, then per-regions
- Render images
- Fix task data displaying with keys different from tag names
- Style improvements

<img width="1190" alt="Screenshot 2025-06-09 at 15 06 32" src="https://github.com/user-attachments/assets/c917793e-d84f-44ee-b38e-b7abebc90687" />

Commit:
fix: BROS-49: TaskSummary postfixes

Commit:
Fix issue on reviewer screen

Commit:
fmt

Commit:
Align to the middle

Commit:
Exclude draft annotations

Commit:
Add test

Commit:
Avoid mocking integrated components in tests

hlomzik committed:
Fix alignment in Labeling Summary

All cells are aligned to top for now.

hlomzik committed:
Fix Pdf tag

hlomzik committed:
Don't filter out empty annotations

This is controlled by setting, so if they are allowed we should show them.
Also tooltips are updated to clarify drafts and predictions situation.

hlomzik committed:
Improve styles

hlomzik committed:
Fix linting for test

hlomzik committed:
Switch to annotation on click in labeling summary

hlomzik committed:
Use react-table for LabelingSummary

Simple version with just a static table and correct renderer wrappers.
Styles are fixed a little to scroll only table, not the whole page.

hlomzik committed:
Add react-table dependency

hlomzik committed:
Also apply scrolling styles to DataSummary

hlomzik committed:
Fix linting

hlomzik committed:
Fix render for labeling type not supported for now

It should just display the number of results

hlomzik committed:
Allow to resize columns in LabelingSummary

hlomzik committed:
Fix linting

hlomzik committed:
Improve styles for resize handler

Also fix table width to exactly match the content

niklub commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15475473690)
> Successfully merged:  rename web/libs/{datamanager/src => core/src/lib}/utils/urlJSON.ts (100%)

niklub committed:
Merge branch 'develop' into 'fb-bros-49'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15475473690

hlomzik committed:
Merge branch 'fb-bros-49' of github.com:HumanSignal/label-studio into fb-bros-49

hlomzik committed:
Use react-table for DataSummary

hlomzik committed:
Improve displaying of TimeSeries data

hlomzik committed:
Cut off long multiline texts at 6 lines

hlomzik committed:
Fix table styles a little

Overflow to avoid weird corners in header; less classnames for cells

hlomzik committed:
Fix results in predictions

They had from_names with postfixes, so all were invalid

hlomzik committed:
Fix styles a little for labels and choices

- wrap long lists of labels
- fix margin between choices

hlomzik committed:
Increase min-width

hlomzik committed:
Always sort labels like in config

Build map of labels from config with enriched color info,
then count labels from result and filter out empty labels after this.
More or less performant iteration through 2 parallel structures.

hlomzik committed:
Show agreement by default for now; show 0-2 decimals if needed

Roles and settings management will be added later.

hlomzik committed:
Rename old View All tab from "Grid" to "Compare"

hlomzik committed:
Sort labelings for easier analysis

Show global classifications first, then labels, then per-regions

hlomzik committed:
Add comments to utils

hlomzik committed:
Fix tsconfig to support `Object.groupBy()`

It's supported from 2023-2024!

hlomzik committed:
Show proper data from real tags

Tag `name` is the key, not task data key

hlomzik committed:
Render images

hlomzik committed:
Render rating

hlomzik committed:
Fix linting

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/DataSummary.tsx:
```diff
@@ -1,32 +1,113 @@
+              key={headerGroup.id}
+              className={[
+                "flex *:flex-shrink-0 *:px-4 *:py-2 bg-neutral-surface",
+                "*:overflow-hidden *:text-ellipsis *:text-left *:whitespace-nowrap",
```
liken in regular css, we don't usually use `*` selector as it's too broad even when targeting nested elements

`*` here will target _every_ element inside the block, even deeply nested. i doubt this is the desired behavior

it's better to use `[&_.selector]:px-4` (`_` here represents space, so it's a selector like `& .selector`) or `[&>.selector]:px-4` to target specific elements

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/DataSummary.tsx:
```diff
@@ -1,32 +1,113 @@
+              key={row.id}
+              className={[
+                "flex *:flex-shrink-0 even:bg-neutral-surface [&_td]:align-top *:px-4 *:py-2",
+                "*:overflow-hidden *:text-ellipsis",
```
same as the above

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15537328252)
> Successfully merged:  create mode 100644 web/libs/editor/tests/integration/e2e/timeseries/multichannel.cy.ts

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/LabelingSummary.tsx:
```diff
@@ -1,57 +1,149 @@
+              key={headerGroup.id}
+              className={[
+                "flex *:flex-shrink-0 *:px-4 *:py-2 bg-neutral-surface",
+                "*:overflow-hidden *:text-ellipsis *:text-left *:whitespace-nowrap",
```
same `*` selector issue

hlomzik committed:
Merge branch 'develop' into 'fb-bros-49'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15537328252

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/LabelingSummary.tsx:
```diff
@@ -1,57 +1,149 @@
+              key={row.id}
+              className={[
+                "flex *:flex-shrink-0 even:bg-neutral-surface [&_td]:align-top *:px-4 *:py-2",
+                "*:overflow-hidden *:text-ellipsis",
```
same `*` selector issue

hlomzik committed:
Revert TS libs list to fix CI

hlomzik committed:
Merge branch 'fb-bros-49' of github.com:HumanSignal/label-studio into fb-bros-49

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/ResizeHandler.tsx:
```diff
@@ -0,0 +1,22 @@
+import type { Header } from "@tanstack/react-table";
```
```suggestion
import type { Header } from "@tanstack/react-table";
import { cnm } from "@humansignal/ui";
```

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/ResizeHandler.tsx:
```diff
@@ -0,0 +1,22 @@
+        "after:content-[''] after:absolute after:bg-neutral-border",
+        "after:right-0 after:top-0 after:h-full after:w-[1px] hover:after:w-[2px]",
+        "cursor-col-resize select-none touch-none",
+      ].join(" ")}
```
we have a special utility for merging classes. it will also properly merge tailwind classes for optimization purposes

```suggestion
      className={cnm(
        "absolute right-0 top-0 h-full w-[4px]",
        "after:content-[''] after:absolute after:bg-neutral-border",
        "after:right-0 after:top-0 after:h-full after:w-[1px] hover:after:w-[2px]",
        "cursor-col-resize select-none touch-none",
      )}
```

nick-skriabin commented on web/libs/editor/src/components/TaskSummary/labelings.tsx:
```diff
@@ -73,11 +62,14 @@ export const renderers: Record<string, RendererType> = {
+            className={[
+              "inline-block whitespace-nowrap rounded-4 px-2",
+              "bg-primary-background border border-primary-emphasis text-accent-grape-dark",
+            ].join(" ")}
```
use `cnm` as in the suggestion above

hlomzik committed:
Fix typo

hlomzik committed:
Improve tailwind usage

- use `cnm()` for long classnames list
- use more precise selectors for direct children: `[&>*]`

hlomzik committed:
Remove unused func params

hlomzik committed:
Improve styles of resize handler

Hit area is bigger and on both sides now

hlomzik commented:
@nick-skriabin applied your suggestions, thank you!
